src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1516,6 +1546,13 @@ CHIP_ERROR CASESession::SendSigma3c(SendSigma3Data & data, CHIP_ERROR status)
     return err;
 }
 
+void CASESession::OnSigma3_AsyncFailure(CHIP_ERROR status)
+{
+    SendStatusReport(mExchangeCtxt, kProtocolCodeInvalidParam);
```

Comment:
How can all this occur on the background thread, possibly while the foreground thread is also potentially operating on the same CASESession object?